src/integrations/grid-eye/grid-eye.service.ts

```diff
@@ -105,7 +106,33 @@ export class GridEyeService
       temperatures.push(await this.getPixelTemperature(i));
     }
 
-    return math.reshape(temperatures, [8, 8]) as number[][];
+    let grid = math.reshape(temperatures, [8, 8]) as number[][];
+    if (this.config.maskZeroBasedValues) {
+      grid = await this.maskZeroBasedValues(grid);
+    }
+    return grid;
+  }
+
+  /**
+   * Replace 0 based values (0 t0 1) with nearest preceding valid value.
+   *
+   * @returns 8x8 matrix of temperatures
+   */
+  async maskZeroBasedValues(temperatures: number[][]): Promise<number[][]> {
+    let goodValue = math.mean(temperatures);
+    for (const [y, row] of temperatures.entries()) {
+      for (const [x, value] of row.entries()) {
+        if (value >= 1) {
+          goodValue = value;
+        } else {
+          row[x] = goodValue;
+          this.logger.debug(
+            `replacing [${x}, ${y}] ${value} with ${goodValue}`
+          );
+        }
+      }
+    }
+    return temperatures;
```

Comment:
This might lead to unexpected behavior during presence detection - if you have a high temperature at [0, 7] and an erroneous reading at [1,0] the high temp would be copied over, resulting in a phantom detection. It's probably quite an edge case, but below I came up with a suggestion that uses a filtered mean to replace the values instead. I also updated the implementation to match the comment on the method (i.e. only replace 0 <= v < 1 instead of everything below 1). Either condition is fine for me, it should just be consistent in documentation and implementation. :)

```suggestion
    const correctedMean = math.mean(temperatures.flat().filter(v => v >= 1 || v < 0))

    return math.matrix(temperatures)
      .map(v => v >= 1 || v < 0 ? v : correctedMean)
      .toArray() as number[][];
```

Would be interested to hear your thoughts on this!